wrapper/src/main/java/software/amazon/jdbc/DriverConnectionProvider.java

```diff
@@ -117,12 +122,71 @@ public Connection connect(
 
     final Properties copy = PropertyUtils.copyProperties(props);
     dialect.prepareConnectProperties(copy, protocol, hostSpec);
+
     final ConnectInfo connectInfo = targetDriverDialect.prepareConnectInfo(protocol, hostSpec, copy);
 
     LOGGER.finest(() -> "Connecting to " + connectInfo.url
         + PropertyUtils.logProperties(PropertyUtils.maskProperties(connectInfo.props), "\nwith properties: \n"));
 
-    Connection conn = this.driver.connect(connectInfo.url, connectInfo.props);
+    Connection conn;
+    try {
+      conn = this.driver.connect(connectInfo.url, connectInfo.props);
+
+    } catch (Throwable throwable) {
+
+      if (!PropertyDefinition.ENABLE_GREEN_NODE_REPLACEMENT.getBoolean(props)) {
```

Comment:
What do you think of moving the duplicated code into a util class or helper method?
```java
  public static HostSpec replaceHost(Throwable throwable, RdsUtils rdsUtils, HostSpec hostSpec, Properties props) throws Throwable {
    if (!PropertyDefinition.ENABLE_GREEN_NODE_REPLACEMENT.getBoolean(props)) {
      throw throwable;
    }

    UnknownHostException unknownHostException = null;
    int maxDepth = 100;
    Throwable loopThrowable = throwable;
    while (--maxDepth > 0 && loopThrowable != null) {
      if (loopThrowable instanceof UnknownHostException) {
        unknownHostException = (UnknownHostException) loopThrowable;
        break;
      }
      loopThrowable = loopThrowable.getCause();
    }

    if (unknownHostException == null) {
      throw throwable;
    }

    if (!rdsUtils.isRdsDns(hostSpec.getHost()) || !this.rdsUtils.isGreenInstance(hostSpec.getHost())) {
      throw throwable;
    }

    // check DNS for such green host name
    InetAddress resolvedAddress = null;
    try {
      resolvedAddress = InetAddress.getByName(hostSpec.getHost());
    } catch (UnknownHostException tmp) {
      // do nothing
    }

    if (resolvedAddress != null) {
      // Green host DNS exists
      throw throwable;
    }

    // Green host DNS doesn't exist. Try to replace it with the corresponding hostname and connect again.

    final String fixedHost = rdsUtils.removeGreenInstancePrefix(hostSpec.getHost());
    return new HostSpecBuilder(hostSpec.getHostAvailabilityStrategy())
        .copyFrom(hostSpec)
        .host(fixedHost)
        .build();
  }
```

Reply:
We can relocate this code to a helper class. The main question whether any other places that require such functionality. I don't see any other place in the driver codebase that may require such checks.